policyengine_us/variables/gov/states/mt/tax/income/mt_taxable_social_security.py

```diff
@@ -0,0 +1,79 @@
+from policyengine_us.model_api import *
+
+
+class mt_taxable_social_security(Variable):
+    value_type = float
+    entity = Person
+    label = "Montana taxable social security benefits"
+    defined_for = StateCode.MT
+    definition_period = YEAR
+    reference = "https://mtrevenue.gov/wp-content/uploads/mdocs/form%202%202021.pdf#page=6"
+
+    def formula(person, period, parameters):
+        p_amount = parameters(
+            period
+        ).gov.states.mt.tax.income.social_security.amount
+        p_rate = parameters(period).gov.irs.social_security.taxability.rate
+        filing_status = person.tax_unit("filing_status", period)
+        # Compute the amount based on the schedule in tax form.
+        # line 1&2: 6a in tax form
+        social_security = person("social_security", period)
+        social_security_benefits_fraction = social_security * p_rate.lower
+        # line 3: irs_gross_income - taxable_social_security
+        taxable_ss = person("taxable_social_security", period)
+        gross_income = person("irs_gross_income", period)
+        reduced_gross_income = max_(gross_income - taxable_ss, 0)
+
+        # line 4: Additions Schedule line 15 - Additions Schedule line 3  (
+        # Iterest and mutual fund dividends from state, county, or municipal bonds
+        # from other states - not included
+        # line 5: tax exempt intrest income
+        tax_emempt_interest_income = person(
+            "tax_exempt_interest_income", period
+        )
+        # line 6: Sum of line 2, 3, 4, 5
+        income_increased_by_ss_and_interest = (
+            social_security_benefits_fraction
+            + reduced_gross_income
+            + tax_emempt_interest_income
+        )
+        # line 7: Above the line deductions less stundet loans &8
+        ald_less_student_loan = person(
```

Comment:
should this be capped at 0?

Reply:
No need, stduent loan interest is part of the ald deduction.